src/command_modules/azure-cli-natgateway/azure/cli/command_modules/natgateway/custom.py

```diff
@@ -0,0 +1,23 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def natgateway_client_factory(cli_ctx, **kwargs):
```

Comment:
This is really still just a generic `network_client_factory`. If you make it end with `.nat_gateways` then it would let you simplify some of your code.